test/sqllogictest/mz_introspection_index_accounting.slt

```diff
@@ -0,0 +1,478 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# The purpose of these tests is to dectect additions or changes of indexes
+# on the mz_introspection cluster, including changes to underlying
+# objects that are indexed.
+# Rewrite this file automatically via --rewrite-results when it fails.
+
+mode cockroach
+
+statement ok
+SET CLUSTER TO mz_introspection
+
+# Record all indexes on the mz_introspection cluster and their definition.
+# That way we can detect new or modified indexes.
+query TT
+SELECT DISTINCT name, create_sql
+FROM mz_indexes
+WHERE cluster_id =
+  (SELECT id FROM mz_clusters WHERE name = 'mz_introspection')
```

Comment:
I think you want to stick an `ORDER BY` here or `rowsort`, otherwise you're depending on whatever arbitrary order this version of Materialize is enforcing, which might be different in future versions!